tests/Feature/SnmpTraps/UpsTrapsOnBatteryTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * UpsTrapsOnBatteryTest.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @author     TheGreatDoc
+ */
+
+namespace LibreNMS\Tests\Feature\SnmpTraps;
+
+use App\Models\Device;
+use App\Models\Sensor;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use LibreNMS\Snmptrap\Dispatcher;
+use LibreNMS\Snmptrap\Trap;
+
+class UpsTrapsOnBatteryTest extends SnmpTrapTestCase
+{
+    public function testOnBattery()
+    {
+        $device = factory(Device::class)->create();
+        $state = factory(Sensor::class)->make(['sensor_class' => 'state', 'sensor_type' => 'upsOutputSourceState', 'sensor_current' => '2']);
+        $time = factory(Sensor::class)->make(['sensor_class' => 'runtime', 'sensor_index' => '100', 'sensor_type' => 'rfc1628', 'sensor_current' => '0']);
+        $remaining = factory(Sensor::class)->make(['sensor_class' => 'runtime', 'sensor_index' => '200', 'sensor_type' => 'rfc1628', 'sensor_current' => '371']);
+        $device->sensors()->save($state);
+        $device->sensors()->save($time);
+        $device->sensors()->save($remaining);
+
+        $trapText = "$device->hostname
+UDP: [$device->ip]:161->[192.168.5.5]:162
+DISMAN-EVENT-MIB::sysUpTimeInstance 9:22:15:00.01
+SNMPv2-MIB::snmpTrapOID.0 UPS-MIB::upsTraps.0.1
+UPS-MIB::upsEstimatedMinutesRemaining.0 100 minutes
+UPS-MIB::upsSecondsOnBattery.0 120 seconds
+UPS-MIB::upsConfigLowBattTime.0 1 minutes";
+
+        \Log::shouldReceive('warning')->never()->with("Snmptrap UpsTraps: Could not find matching sensor \'Estimated battery time remaining\' for device: " . $device->hostname);
```

Comment:
why are you using never()?  This actually changes it to shouldNotRecieve()...

Reply:
Actually, I took that example from the jnx* trap tests, for example here https://github.com/librenms/librenms/blob/300645388fefdcd999bc6dff4738b08959c6abac/tests/Feature/SnmpTraps/JnxBgpM2Test.php#L61
But I dont mind changin them to shouldNotReceive()

Reply:
@murrant I've changed without looking and shouldNotReceive doesnt exists...